### DISCUSSION TITLE
HMR update for multiple assets

### DISCUSSION THREAD

**Discussion by 101arrowz** [parcel-bundler/parcel#5317] [Q&A] [answered]
The following is in the [HMR runtime](https://github.com/parcel-bundler/parcel/blob/v2/packages/runtimes/hmr/src/loaders/hmr-runtime.js#L61-L85):
```js
assets.forEach(asset => {
  var didAccept =
    asset.type === 'css' || hmrAcceptCheck(module.bundle.root, asset.id);
  if (didAccept) {
    handled = true;
  }
});
```
I gather that this means that if *any* of the assets can accept an HMR update, then we do not need to reload. However, I would think that for an HMR update to work, you should ensure that *all* assets can accept it. Am I missing something?

**Comment by mischnic** [comment, marked as answer]
Yeah. I think in Webpack, these updates also somehow bubble and if no parent accepts it, then the page reloads. (And generally, we should do what Webpack does)

**Reply by 101arrowz** [reply]
That seems logically flawed to me; it makes it possible to force HMR to try to apply for an asset change that would normally not work by simply changing a CSS asset at the same time. It's possible this is a bug that never surfaced because it's effectively impossible to modify two assets simultaneously as a developer. At the same time, I'm not as knowledgeable as the Webpack authors, so I suppose it should be left alone for now...

**Reply by mischnic** [reply]
> to modify two assets simultaneously as a developer

You could get multiple assets though if you add an import for example. But that's not really modifying 

> At the same time, I'm not as knowledgeable as the Webpack authors, so I suppose it should be left alone for now...

Ideally, we would run the Webpack HMR test suite on Parcel...

**Reply by 101arrowz** [reply]
You'd have to modify the behavior of `module.hot.accept` to be compliant with Webpack HMR. I think I mentioned that in #4935, and I worked a bit in [this branch](https://github.com/101arrowz/parcel/tree/webpack-hmr-api).
